src/swap/sideshift.js

```diff
@@ -0,0 +1,301 @@
+// @flow
+
+import { gt, lt } from 'biggystring'
+import {
+  SwapAboveLimitError,
+  SwapBelowLimitError,
+  SwapPermissionError
+} from 'edge-core-js'
+import { EdgeFetchResponse } from 'edge-core-js/lib/types/types'
+import {
+  type EdgeCorePluginOptions,
+  type EdgeCurrencyWallet,
+  type EdgeSpendInfo,
+  type EdgeSwapInfo,
+  type EdgeSwapPlugin,
+  type EdgeSwapQuote,
+  type EdgeSwapRequest,
+  type EdgeTransaction,
+  SwapCurrencyError
+} from 'edge-core-js/types'
+
+import { makeSwapPluginQuote } from '../swap-helpers.js'
+
+// const INVALID_CURRENCY_CODES = {} // TODO: anything to add here?
+
+// Invalid currency codes should *not* have transcribed codes
+// because currency codes with transcribed versions are NOT invalid
+const CURRENCY_CODE_TRANSCRIPTION = {
+  // Edge currencyCode: exchangeCurrencyCode
+  // TODO: any other transcription neeeded?
+  USDT: 'usdtErc20'
+}
+
+const pluginId = 'sideshift'
+const swapInfo: EdgeSwapInfo = {
+  pluginId,
+  displayName: 'SideShift.ai',
+  supportEmail: 'help@sideshift.ai'
+}
+type FixedQuote = {
+  createdAt: string,
+  depositAmount: string,
+  depositMethod: string,
+  expiresAt: string,
+  id: string,
+  rate: string,
+  settleAmount: string,
+  settleMethod: string
+}
+
+type FixedQuoteRequestParams = {
+  depositMethod: string,
+  settlMethod: string,
+  depositAmount: string
+}
+
+type OrderRequest = {
+  createdAt: string,
+  createdAtISO: string,
+  expiresAt: Date,
+  expiresAtIso: string,
+  depositAddress: {
+    address: string
+  },
+  depositMethod: string,
+  id: string,
+  orderId: string,
+  settleAddress: {
+    address: string
+  },
+  settleMethod: string,
+  depositMax: string,
+  depositMin: string,
+  quoteId: string,
+  settleAmount: string,
+  depositAmount: string,
+  deposits: Array<any>
+}
+
+type OrderRequestParams = {
+  type: string,
+  quoteId: string,
+  affiliateId: string,
+  sessionSecret?: string,
+  settleAddress: string
+}
+
+type Rate = {
+  rate: number,
+  min: string,
+  max: string,
+  error?: {
+    message: string
+  }
+}
+
+type Permission = {
+  createOrder: boolean
+}
+
+const dontUseLegacy = {
+  DGB: true
```

Comment:
DGB? Is this what I think it means?! 👀

Reply:
removing this part, since we support legacy addresses + we don't support DGB anyway.

Reply:
I've kept the dontUseLegacy variable in the code as an empty object, since we want Edge to point others to our code as a reference, I believe it is best to keep it there.